tests/filesystem/conftest.py

```diff
@@ -0,0 +1,83 @@
+from typing import Union, Any, Iterator
+
+import pytest
+
+import os
+import subprocess
+import shutil
+import pathlib
+from .settings import (
+    TEST_SAMPLES_PATH,
+    REPO_FIXTURE_PATH,
+    REPO_SAFE_PREFIX,
+    REPO_GOOD_REF,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def repo_fixture(
+    repo_path: Union[str, os.PathLike] = REPO_FIXTURE_PATH,
+    samples_path: Union[str, os.PathLike] = TEST_SAMPLES_PATH,
+    safe_prefix: str = REPO_SAFE_PREFIX,
+    good_ref: str = REPO_GOOD_REF,
+) -> Iterator[Any]:
+    """Create a temporary git repository to test git-based filesystems.
+
+    Args:
+        repo_path (str): The path at which to create the temporary repo. Defaults to REPO_FIXTURE_PATH. It is safest
+            to create the repo outside your software project's directory tree so it does not interfere with real repos.
+        samples_path (str): The path to the sample files, which will be added to the repo. Defaults to TEST_SAMPLES_PATH.
+        safe_prefix (str): Helps prevent mixups between the test repo and real repos. Defaults to REPO_SAFE_PREFIX.
+        good_ref (str): The git ref for the unmodified sample files. Later commits intentionally break the sample
+            files so that tests will fail if the system under test doesn't correctly handle refs.
+            Defaults to REPO_SAFE_REF.
+
+    Yields:
+        Tuple[str, str]: A tuple containing the repo_path and good_ref.
+
+    """
+    repo_path = pathlib.Path(repo_path)
+    samples_path = pathlib.Path(samples_path)
+
+    try:
+        try:
+            os.mkdir(repo_path)
+        except FileExistsError:
+            raise FileExistsError(
+                f"Directory `{repo_path.absolute()}` already exists."
+                "It should have been removed by the previous test run."
+            )
+
+        # NOTE: `git init -b` option requires git 2.28 or later.
+        subprocess.call(f"git init -b {safe_prefix}master", shell=True, cwd=repo_path)
```

Comment:
Good question.   For gitpython way, do you mean:

1.  just using `git.<command>(...)` instead of the above `subprocess.call()`.  Like this https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly ?
2. Or deeper in the gitpython API?  Oh, I think it's read-only.  

For me `subprocess.call()` is easier to read because it looks like git on the command line.   I don't want the reader to have to learn a new API to understand the test data.     Also it's how I saw test repo done in [other fsspec implementations](https://github.com/fsspec/filesystem_spec/blob/master/fsspec/implementations/tests/test_git.py).

The code lines did get kind of long though ☹️ .  And gitpython would let other settings be made if needed.

I am happy to try the gitpython way.  Let me know.